### PR TITLE
feature: add the build option of the client in the local environment

### PR DESCRIPTION
**Pull request by fengzixu**
Signed-off-by: fengzixu <hnustphoenix@gmail.com>

### Ⅰ. Describe what this PR did

1. add the script to build the client in the local env other than docker
2. add the build option in the Makefile

### Ⅱ. Does this pull request fix one issue?

fixed https://github.com/dragonflyoss/Dragonfly/issues/313

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

Just makefile and a script of building the client

### Ⅳ. Describe how to verify it

Run two commands:

1. cd hack && ./build-client-local.sh
2. make build-client-local

### CODE REVIEW

File: hack/build-client-local.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            build-dfget-local
+        ;;
+        *)
+            build-dfget-local
+            build-dfdaemon-local
+        ;;
+    esac
+}
+
+main "$@"
```
Comment (allencloud) [inline hack/build-client-local.sh]: please wrap the line.

**Comment by fengzixu** [comment]
Is this PR can satisfy your requirement? @allencloud

**Comment by allencloud** [comment]
I tested the branch of this pr, and got the following display:
```
Dragonfly (master) $ git fetch upstream pull/316/head:new-316
remote: Enumerating objects: 3, done.
remote: Counting objects: 100% (3/3), done.
remote: Total 5 (delta 3), reused 3 (delta 3), pack-reused 2
Unpacking objects: 100% (5/5), done.
From github.com:dragonflyoss/dragonfly
 * [new ref]         refs/pull/316/head -> new-316
Dragonfly (master) $ git checkout new-316
Switched to branch 'new-316'
Dragonfly (new-316) $ make build-client-local
Begin to check client code formats.
./hack/check-client.sh
CHECK: gofmt, check code formats
CHECK: golint, check code style
CHECK: go vet, check code syntax
./hack/build-client-local.sh
BUILD: dfget in the local environment.
BUILD: dfdaemon in the local environment.
Dragonfly (new-316) $ which dfget
Dragonfly (new-316) $
```

How about removing all the gofmt, golint, govet checking when building? And explicitly show the location of where the binaries. @fengzixu

File: hack/build-client-local.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+BIN_DIR="../../release"
```
Comment (allencloud) [inline hack/build-client-local.sh]: how about changing it to be `../../bin` and then add `bin` in the .gitignore file? @fengzixu

File: Makefile [hunk 3]
```diff
@@ -104,4 +104,11 @@ uninstall:
 build-client: check-client build-dfget build-dfdaemon
 .PHONY: build-client
 
-build: check-client build-dfget build-dfdaemon check-supernode build-supernode
\ No newline at end of file
+# Create for the tool of generate docs of the client.
```
Comment (allencloud) [inline Makefile]: I am afraid building binaries is very essential for developers. For example, I code on my Mac and have no docker environment, but I still use the Makefile to build binaries to test. 
Thus it is not for doc generation. How about changing that?

File: hack/build-client-local.sh (hunk 2 above)
Comment (fengzixu) [reply hack/build-client-local.sh]: Actually, we have used the dir called release to store temporary binary files. According to your comment, adding release into the .gitignore file is better. WYDT? @allencloud

**Comment by fengzixu** [comment]
I run the command `make build-client-local`

```
➜  Dragonfly git:(master) ✗ make build-client-local
./hack/build-client-local.sh
BUILD: dfget in /Users/xuran/code/go-project/src/github.com/dragonflyoss/Dragonfly/bin/dfget
BUILD: dfdaemon in /Users/xuran/code/go-project/src/github.com/dragonflyoss/Dragonfly/bin/dfdaemon
```
After this PR was merged, I will refactor the makefile as soon as possible. @allencloud

**Comment by allencloud** [comment]
LGTM

**Pull request merged**